pennylane/workflow/execution.py

```diff
@@ -409,38 +398,13 @@ def cost_fn(params, x):
         diff_method, grad_on_execution, interface, device, device_vjp, mcm_config, gradient_kwargs
     )
 
-    is_gradient_transform = isinstance(diff_method, qml.transforms.core.TransformDispatcher)
-    transform_program, inner_transform = _make_transform_programs(
-        device, config, inner_transform, transform_program, is_gradient_transform
-    )
-
-    # If caching is desired but an explicit cache is not provided, use an ``LRUCache``.
-    if cache is True:
-        cache = LRUCache(maxsize=cachesize)
-        setattr(cache, "_persistent_cache", False)
-
-    # Ensure that ``cache`` is not a Boolean to simplify downstream code.
-    elif cache is False:
-        cache = None
-
-    # changing this set of conditions causes a bunch of tests to break.
-    no_interface_boundary_required = interface == "numpy" or config.gradient_method in {
-        None,
-        "backprop",
-    }
-    device_supports_interface_data = no_interface_boundary_required and (
-        interface == "numpy"
-        or config.gradient_method == "backprop"
-        or getattr(device, "short_name", "") == "default.mixed"
-    )
+    # pylint: disable=protected-access
+    if transform_program is None or inner_transform is None:
+        transform_program, inner_transform, config = qml.workflow._setup_transform_program(

```

Comment:
Can we instead do `from .qnode import _setup_transform_program`?

If that is causing circular import issues, we should probably already move it to a separate file.

Reply:
Yeah, that's the plan. Today I will move it to its own file and add some tests for `_setup_transform_program` and `_resolve_execution_config`.